tuxedo-template-runtime/src/lib.rs

```diff
@@ -159,17 +159,26 @@ impl BuildStorage for GenesisConfig {
             WASM_BINARY.unwrap().to_vec(),
         );
 
+        use sp_api::HashT;
         use tuxedo_core::inherents::InherentInternal;
-        let mut genesis_utxos: Vec<Output> = OuterConstraintCheckerInherentHooks::genesis_utxos();
-        genesis_utxos.extend(self.genesis_utxos.clone());
 
-        for (index, utxo) in genesis_utxos.iter().enumerate() {
-            let output_ref = OutputRef {
-                // Genesis UTXOs don't come from any real transaction, so just use the zero hash
-                tx_hash: <Header as sp_api::HeaderT>::Hash::zero(),
-                index: index as u32,
-            };
-            storage.top.insert(output_ref.encode(), utxo.encode());
+        let mut genesis_transactions: Vec<Transaction> =
+            OuterConstraintCheckerInherentHooks::genesis_transactions();
+        genesis_transactions.extend(self.genesis_transactions.clone());
+
+        storage
+            .top
+            .insert(EXTRINSIC_KEY.to_vec(), genesis_transactions.encode());
```

Comment:
The comment at the beginning of this function needs revised now that we are doing interesting things here.